process/block/baseProcess.go

```diff
@@ -854,7 +855,8 @@ func (bp *baseProcessor) requestMissingFinalityAttestingHeaders(
 		if len(headers) == 0 {
 			missingFinalityAttestingHeaders++
 			requestedHeaders++
-			go bp.requestHeaderByShardAndNonce(shardId, i)
+			// TODO change epoch here
+			go bp.requestHeaderByShardAndNonce(shardId, i, 0)
```

Comment:
here you can ask with the epoch of the last block, you do not know if that is with new epoch or not.

Reply:
or with your current epoch.

Reply:
refactored